packages/server-wallet/jest/with-peers-setup-teardown.ts

```diff
@@ -61,96 +58,133 @@ export const participantIdB = 'b';
  * @param enginesToRestart Specifies the peerEngines that will be restarted
  */
 export async function crashAndRestart(
-  enginesToRestart: 'A' | 'B' | 'Both' = 'Both'
-): Promise<void> {
+  oldPeerSetup: PeerSetup,
+  enginesToRestart: 'A' | 'B' | 'Both'
+): Promise<PeerSetup> {
   try {
-    await messageService.destroy();
+    await oldPeerSetup.messageService.destroy();
 
-    if (enginesToRestart === 'A' || enginesToRestart === 'Both') {
-      await peerEngines.a.destroy();
-      peerEngines.a = await Engine.create(aEngineConfig);
-    }
+    const restartA = enginesToRestart === 'A' || enginesToRestart === 'Both';
+    const restartB = enginesToRestart === 'B' || enginesToRestart === 'Both';
 
-    if (enginesToRestart === 'B' || enginesToRestart === 'Both') {
-      await peerEngines.b.destroy();
-      peerEngines.b = await Engine.create(bEngineConfig);
+    if (restartA) {
+      await oldPeerSetup.peerEngines.a.destroy();
+    }
+    if (restartB) {
+      await oldPeerSetup.peerEngines.b.destroy();
     }
+    const a = restartA ? await Engine.create(aEngineConfig) : oldPeerSetup.peerEngines.a;
+
+    const b = restartB ? await Engine.create(bEngineConfig) : oldPeerSetup.peerEngines.b;
 
     const handler = await createTestMessageHandler([
-      {participantId: participantIdA, engine: peerEngines.a},
-      {participantId: participantIdB, engine: peerEngines.b},
+      {participantId: participantIdA, engine: a},
+      {participantId: participantIdB, engine: b},
     ]);
 
-    messageService = (await TestMessageService.create(
+    const participantA = {
+      signingAddress: await a.getSigningAddress(),
+      participantId: participantIdA,
+      destination: makeDestination(
+        '0x00000000000000000000000000000000000000000000000000000000000aaaa1'
+      ),
+    };
+    const participantB = {
+      signingAddress: await b.getSigningAddress(),
+      participantId: participantIdB,
+      destination: makeDestination(
+        '0x00000000000000000000000000000000000000000000000000000000000bbbb2'
+      ),
+    };
+
+    const messageService = (await TestMessageService.create(
```

Comment:
Aside:

Is there a reason that `TestMessageService.create()` doesn't return a `Promise<TestMessageService>` directly? This would still satisfy any signature looking for a `MessageServiceInterface`, but isn't 'lossy'.

Reply:
That's a great idea!

Reply:
Done in 805a0b2